enhancements/etcd/automated-backups-no-config.md

```diff
@@ -0,0 +1,271 @@
+Once the container fails, it will be automatically restarted according to exponential backoff strategy. This scenario leads to `CrashLoopBackOff` and the etcd Pod will never be `Running`. 
+
+As the no-config backup sidecar is being deployed alongside each etcd member, consequently an error within this container leads to unavailable etcd cluster. Hence, failing the container upon error is not a viable option.
+Also disabling the container from the etcd pod upon error is not recommended, unless requested by the user. As discussed above, enabling and disabling the sidecar container leads to deploying a new version of the static pods.
+This scenario leads to a duration of cluster unavailability. 
+
+As shown above, it is necessary to report the no-config backup sidecar failure without failing the container. Below are possible resolution scenarios. 
+
+- Log the error upon failure
+  - A log parser could be used in order to detect the error, without failing the container.
+  
+- Update Cluster-etcd-Operator StatusConditions
+  - Reporting the condition and degrading the operator could be a solution. 
+
+- Use a Prometheus gauge in order to report the container situation. 
+
+### Implementation Details/Notes/Constraints
+
+The _no-config_ backups builds atop of the existing [automated backup of etcd](https://docs.openshift.com/container-platform/4.15/backup_and_restore/control_plane_backup_and_restore/backing-up-etcd.html#creating-automated-etcd-backups_backup-etcd).
+
+The `PeriodicBackupController` within **cluster-etcd-operator** checks for `Backup` CR with `name=default` within every sync. Upon applying a `Backup` CR with `name=default`, the _no-config_ backup sidecar container is being enabled within each etcd pod in Openshift cluster.
+The `Backup` configurations such as `scheudle`, `timezone` and `retention policy`, are being supplied to the _no-config_ container as flags. 
+According to user's preference, the default configuration could be modified by updating the CR.
+
+The backups are being stored on the local disk of each master node independently, using `hostPath` volume. This replication is required for disaster recovery scenarios where more than one of the master nodes has been lost.
+
+
+A `Backup` CR with default configuration below.
+
+```yaml
+apiVersion: config.openshift.io/v1alpha1
+kind: Backup
+metadata:
+  name: default
+spec:
+  etcd:
+    schedule: "*/5 * * * *"
+    timeZone: "UTC"
+    retentionPolicy:
+      retentionType: RetentionNumber
+      retentionNumber:
+        maxNumberOfBackups: 3
+```
+
+
+
+
+### Risks and Mitigations
+
+## Test Plan
+
+An e2e test will be added to practice the scenario as follows 
+
+- Enable AutomatedBackupNoConfig FeatureGate.
+- Verify that a `Backup` CR has been installed.
+- Verify that backups have been taken successfully.
+- Verify that backups are being maintained according to the retention policy.
+
```

Comment:
e2e test should also attempt to restore etcd from backup to make sure backup is not corrupted

Reply:
the restore is out of scope so far iiuc 

cc @openshift/openshift-team-etcd wdyt

Reply:
we have separate restore tests that take a fresh backup and restore from it

Reply:
> e2e test should also attempt to restore etcd from backup to make sure backup is not corrupted


> we have separate restore tests that take a fresh backup and restore from it


We need a check to be sure our backup is valid.  If that can be done without a restore, ok. If not, then wiring up a restore after the backup is taken would work really well.

Reply:
we can instead 

- compare with the backup on other nodes
- compare with the data on the current node

iiuc, the restoration  should be done on a fresh node. If the test will allocate a new master node, the test time will be significantly  long, wdyt !!